Handlers/PricePartHandler.cs

```diff
@@ -1,23 +1,53 @@
+using System;
+using System.Linq;
 using System.Threading.Tasks;
 using OrchardCore.Commerce.Abstractions;
 using OrchardCore.Commerce.Models;
+using OrchardCore.Commerce.Settings;
 using OrchardCore.ContentManagement.Handlers;
+using OrchardCore.ContentManagement.Metadata;
 
 namespace OrchardCore.Commerce.Handlers
 {
     public class PricePartHandler : ContentPartHandler<PricePart>
     {
-        private IMoneyService _moneyService;
+        private readonly IMoneyService _moneyService;
+        private readonly IContentDefinitionManager _contentDefinitionManager;
 
-        public PricePartHandler(IMoneyService moneyService)
+        public PricePartHandler(IMoneyService moneyService, IContentDefinitionManager contentDefinitionManager)
         {
             _moneyService = moneyService;
+            _contentDefinitionManager = contentDefinitionManager;
+        }
+
+        public override Task InitializingAsync(InitializingContentContext context, PricePart part)
+        {
+            GetCurrencySelectionMode(part);
```

Comment:
Why do we need to do that both on initializing and loading?

Reply:
It wasn't needed in both. Removed from Loading.